packages/framework/tests/Feature/Services/Markdown/DynamicMarkdownLinkProcessorTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/** @noinspection HtmlUnknownTarget */
+
+declare(strict_types=1);
+
+namespace Hyde\Framework\Testing\Feature\Services\Markdown;
+
+use Hyde\Pages\InMemoryPage;
+use Hyde\Testing\UnitTestCase;
+use Hyde\Support\Models\Route;
+use Hyde\Support\Facades\Render;
+use Hyde\Foundation\Facades\Routes;
+use Hyde\Support\Models\RenderData;
+use Hyde\Markdown\Processing\DynamicMarkdownLinkProcessor;
+
+/**
+ * @covers \Hyde\Markdown\Processing\DynamicMarkdownLinkProcessor
+ */
+class DynamicMarkdownLinkProcessorTest extends UnitTestCase
+{
+    protected static bool $needsConfig = true;
+    protected static bool $needsKernel = true;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Render::swap(new RenderData());
+
+        Routes::addRoute(new Route(new InMemoryPage('home')));
+    }
+
+    public function testRouteReplacement()
+    {
+        $input = '<p><a href="hyde::route(\'home\')">Home</a></p>';
+        $expected = '<p><a href="home.html">Home</a></p>';
+
+        $this->assertSame($expected, DynamicMarkdownLinkProcessor::postprocess($input));
+    }
+
+    public function testRouteReplacementWithDoubleQuotes()
+    {
+        $input = '<p><a href="hyde::route("home")">Home</a></p>';
+        $expected = '<p><a href="home.html">Home</a></p>';
+
+        $this->assertSame($expected, DynamicMarkdownLinkProcessor::postprocess($input));
+    }
```

Comment:
Double quotes can't be supported since the "src" attribute is using double quotes in the compiled HTML.

Reply:
Unless we add a workaround, we may want to document this reasoning. For now, it's reverted in https://github.com/hydephp/develop/pull/1590/commits/df09449ab2e856f3fe360888382f958cf3c104bf. Keeping thread open for now.